control-plane/pkg/reconciler/kafka/kafka.go

```diff
@@ -0,0 +1,219 @@
+				case "SCRAM-SHA-256":
+					secret.StringData["sasl"] = "scram_sha256"
+				case "SCRAM-SHA-512":
+					secret.StringData["sasl"] = "scram_sha512"
+				case "PLAIN":
+					secret.StringData["sasl"] = "plaintext"
+				default:
+					return nil, nil, fmt.Errorf("SASL type value %q is not supported", *saslType)
+				}
+			} else {
+				secret.StringData["sasl"] = "plaintext" //default
+			}
+
+			sasl := kedav1alpha1.AuthSecretTargetRef{Parameter: "sasl", Name: secret.Name, Key: "sasl"}
+
+			username := kedav1alpha1.AuthSecretTargetRef{
+				Parameter: "username",
+				Name:      cg.Spec.Template.Spec.Auth.NetSpec.SASL.User.SecretKeyRef.Name,
+				Key:       cg.Spec.Template.Spec.Auth.NetSpec.SASL.User.SecretKeyRef.Key,
+			}
+			password := kedav1alpha1.AuthSecretTargetRef{
+				Parameter: "password",
+				Name:      cg.Spec.Template.Spec.Auth.NetSpec.SASL.Password.SecretKeyRef.Name,
+				Key:       cg.Spec.Template.Spec.Auth.NetSpec.SASL.Password.SecretKeyRef.Key,
+			}
+
+			secretTargetRefs = append(secretTargetRefs, sasl, username, password)
+			triggerAuth.Spec.SecretTargetRef = secretTargetRefs
+		}
+
+		if cg.Spec.Template.Spec.Auth.NetSpec.TLS.Enable {
+			secret.StringData["tls"] = "enable"
+			tls := kedav1alpha1.AuthSecretTargetRef{Parameter: "tls", Name: secret.Name, Key: "tls"}
+			secretTargetRefs = append(secretTargetRefs, tls)
+
+			if cg.Spec.Template.Spec.Auth.NetSpec.TLS.CACert.SecretKeyRef != nil {
+				ca := kedav1alpha1.AuthSecretTargetRef{
+					Parameter: "ca",
+					Name:      cg.Spec.Template.Spec.Auth.NetSpec.TLS.CACert.SecretKeyRef.Name,
+					Key:       cg.Spec.Template.Spec.Auth.NetSpec.TLS.CACert.SecretKeyRef.Key,
+				}
+
+				secretTargetRefs = append(secretTargetRefs, ca)
+			}
+
+			if cg.Spec.Template.Spec.Auth.NetSpec.TLS.Cert.SecretKeyRef != nil {
+				cert := kedav1alpha1.AuthSecretTargetRef{
+					Parameter: "cert",
+					Name:      cg.Spec.Template.Spec.Auth.NetSpec.TLS.Cert.SecretKeyRef.Name,
+					Key:       cg.Spec.Template.Spec.Auth.NetSpec.TLS.Cert.SecretKeyRef.Key,
+				}
+				secretTargetRefs = append(secretTargetRefs, cert)
+			}
+
+			if cg.Spec.Template.Spec.Auth.NetSpec.TLS.Key.SecretKeyRef != nil {
+				key := kedav1alpha1.AuthSecretTargetRef{
+					Parameter: "key",
+					Name:      cg.Spec.Template.Spec.Auth.NetSpec.TLS.Key.SecretKeyRef.Name,
+					Key:       cg.Spec.Template.Spec.Auth.NetSpec.TLS.Key.SecretKeyRef.Key,
+				}
+				secretTargetRefs = append(secretTargetRefs, key)
+			}
+			triggerAuth.Spec.SecretTargetRef = secretTargetRefs
+		}
+	}
+
+	if cg.Spec.Template.Spec.Auth.AuthSpec != nil && cg.Spec.Template.Spec.Auth.AuthSpec.Secret.Ref.Name != "" {
+		host := kedav1alpha1.AuthSecretTargetRef{
+			Parameter: "host",
+			Name:      cg.Spec.Template.Spec.Auth.AuthSpec.Secret.Ref.Name,
+			Key:       "", //TODO: add key
+		}
+		secretTargetRefs = append(secretTargetRefs, host)
+		triggerAuth.Spec.SecretTargetRef = secretTargetRefs
+		return triggerAuth, nil, nil
+	}
```

Comment:
in this case we have well-known keys to use, documented at https://knative.dev/docs/eventing/brokers/broker-types/kafka-broker/#security

Reply:
Since in the `AuthSpec` case, we aren't using any of the authentication methods with keys listed in that doc and only have a [SecretReference](https://github.com/knative-sandbox/eventing-kafka-broker/blob/main/control-plane/pkg/apis/eventing/v1alpha1/kafka_sink_types.go#L116-L119), we'll need a new/custom parameter/key. I can leave this as `to be determined`.